### PR TITLE
Playlist Index Endpoint

### LINKED ISSUE #10: Backend: Playlist Index Request
GET /api/v1/playlists

Returns all the playlists in the database along with their associated favorited songs

If successful, this request will return a response in the following format:
```
[
    {
        "id": 1,
        "playlist_name": "Favorite songs of all time",
        "favorites": [
          {
            "id": 1,
            "name": "We Will Rock You",
            "artist_name": "Queen"
            "genre": "Rock",
            "rating": 88
          },
          {
            "id": 2,
            "name": "Careless Whisper",
            "artist_name": "George Michael"
            "genre": "Pop",
            "rating": 93
          }
        ]
    },
    {
        "id": 2,
        "name": "Other amazing songs",
        "favorites": [
          {
            "id": 1,
            "name": "We Will Rock You",
            "artist_name": "Queen"
            "genre": "Rock",
            "rating": 88
          },
          {
            "id": 2,
            "name": "Careless Whisper",
            "artist_name": "George Michael"
            "genre": "Pop",
            "rating": 93
          },
        ]
    },
]
```

### PR DESCRIPTION
**Pull request by cebarks**
closes #10 

- adds 'cors' and 'pryjs' dependencies
- adds testing for favorites index endpoint
- implements favorites index endpoint

### CODE REVIEW

File: db/migrations/20190320104850_initial.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ exports.up = function(knex, Promise) {
     }),
     knex.schema.createTable('playlists_favorites', (table) => {
       table.increments('id').primary();
-      table.integer('favorite_id').references('favorites.id');
-      table.integer('playlist_id').references('playlists.id');
+      table.integer('favorite_id').references('favorites.id').onDelete('CASCADE');
+      table.integer('playlist_id').references('playlists.id').onDelete('CASCADE');
```
Comment (bendelonlee) [inline db/migrations/20190320104850_initial.js:23]: So basically, dependent destroy?

File: index.js [hunk 2]
```diff
@@ -66,6 +68,26 @@ app.delete('/api/v1/favorites/:id', (request, response) => {
     });
 });
 
+app.get('/api/v1/playlists', (request, response) => {
+  database('playlists')
+  .innerJoin('playlists_favorites', 'playlists.id', 'playlists_favorites.playlist_id')
+  .innerJoin('favorites', 'favorites.id', 'playlists_favorites.favorite_id')
+  .select(['playlists.id', 'playlists.title', database.raw("JSON_AGG(favorites) as favorites")])
+  .groupBy('playlists.id', 'playlists_favorites.id', 'favorites.id')
+  .then(playlists => {
```
Comment (bendelonlee) [inline index.js:97]: Very well done working out this advanced query.

File: seeds/seed.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  return knex('playlists_favorites').del()
+  .then(() => knex('playlists').del())
+  .then(() => knex('favorites').del())
+  .then(() => {
+    return Promise.all([
+      knex('favorites').insert(newFavorites,'id')
+      .then((favorites) => {
+        return knex('playlists').insert(newPlaylists, 'id')
+        .then((playlists) => {
+          return knex('playlists_favorites').insert([
```
Comment (bendelonlee) [inline seeds/seed.js:23]: I'm not entirely sure what the `return` is doing here. Does it break when you remove it?

**Review by bendelonlee — COMMENTED** [review]
I left some comments, please look at them, though I'm not requesting any changes.

File: db/migrations/20190320104850_initial.js (hunk 1 above)
Comment (cebarks) [reply db/migrations/20190320104850_initial.js:23]: yup exactly, just implemented by postgres instead of ActiveRecord

File: index.js (hunk 2 above)
Comment (cebarks) [reply index.js:97]: Thanks to Maddie for the help!

File: seeds/seed.js (hunk 3 above)
Comment (cebarks) [reply seeds/seed.js:23]: From my understanding, the return is required when you're chaining Promise's together.

**Pull request closed without merging**